core/shared/src/main/scala/zio/internal/macros/LayerBuilder.scala

```diff
@@ -114,13 +115,28 @@ final case class LayerBuilder[Type, Expr](
    * confusing the user and breaking stuff.
    */
   private def assertNoAmbiguity(): Unit = {
-    val typesToExprs: Map[String, List[String]] =
-      groupMap(providedLayerNodes.flatMap { node =>
-        node.outputs.map(output => showType(output) -> showExpr(node.value))
-      })(_._1)(_._2)
+    val simpleDuplicates: Map[String, List[String]] =
+      groupMap {
+        providedLayerNodes.flatMap { node =>
+          node.outputs.map(output => showType(output) -> showExpr(node.value))
+        }
+      }(_._1)(_._2)
+
+    val subtypingDuplicates: Map[String, List[String]] =
+      groupMap {
+        (target ++ providedLayerNodes.flatMap(_.inputs))
+          .filterNot(typeEquals(anyType, _))
```

Comment:
This feels like a stupid hack, but without it the code in this repo (and I assume in many other places) won't compile. It fails with an ambiquity error `zio.ZAny is provided by:` which looks like some Scala type inference issue, since `type ZAny >: Any`. 
So here we just throw away any type for which `Any <:< tpe` is true (including `Any` apparently).